client/jetpack-cloud/sections/agency-dashboard/hooks.tsx

```diff
@@ -65,3 +68,169 @@ export function useToggleActivateMonitor( {
 
 	return [ toggle, toggleActivateMonitoring.isLoading ];
 }
+
+export function useUpdateMonitorSettings( sites: Array< { blog_id: number; url: string } > ): {
+	updateMonitorSettings: ( params: object ) => void;
+	isLoading: boolean;
+	isComplete: boolean;
+} {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const queryClient = useQueryClient();
+	const { filter, search, currentPage } = useContext( SitesOverviewContext );
+	const queryKey = [ 'jetpack-agency-dashboard-sites', search, currentPage, filter ];
+
+	const [ status, setStatus ] = useState( 'idle' );
+
+	const updateMonitorSettings = useUpdateMonitorSettingsMutation( {
+		onSuccess: async ( data, { siteId } ) => {
+			// Cancel any current refetches, so they don't overwrite our optimistic update
+			await queryClient.cancelQueries( queryKey );
+
+			// Snapshot the previous value
+			const previousSites = queryClient.getQueryData( queryKey );
+
+			// Optimistically update to the new value
+			queryClient.setQueryData( queryKey, ( oldSites: any ) => {
+				return {
+					...oldSites,
+					sites: oldSites?.sites.map( ( site: Site ) => {
+						if ( site.blog_id === siteId ) {
+							return {
+								...site,
+								monitor_settings: {
+									...site.monitor_settings,
+									monitor_deferment_time: data.settings.jetmon_defer_status_down_minutes,
```

Comment:
We are optimistically updating all the values, but `email_notifications` & `wp_note_notifications` will be effective once the API changes are done.

Reply:
Is this the reason why the email address I entered isn't saved?

Reply:
Yes, correct.